### ISSUE TITLE
Combine infrastructure simplification

### ISSUE THREAD

**Issue by derekjn** [pipelinedb/pipelinedb#1080]
This refs #1068 but I want to get it into master before #1068 is complete because this probably affects #729.

Basically I think we need to unify everything that happens with creating transition state columns so that any continuous query returned by the analyzer always has the exact same `TupleDesc` as the matrel. Currently we create transition state columns outside of the analyzer in `pipelinecmds.c:ExecCreateContViewStmt` and perform a bit of sorcery in `cont_plan.c:set_plan_refs` (and probably a couple of other places).

One of the things that makes `cont_plan.c:set_plan_refs` complex is that it has to deal with aligning subtly different `TupleDescs` between the worker and combiner. The main difference is that combiner plans have a "store" target, which is the finalized aggregate value.

But I don't think we need to store finalized aggregate values anywhere because the continuous view's overlay can just call `combine` on the stored transition state, which is equivalent to finalizing an aggregate. 

We just need a transition state extraction function and another form of `combine` that takes a function and a column name, and returns the combined transition state. Then, the combiner would simply rewrite all aggregates to `combines` on the output from the worker. So something like this would work:

    -- worker
    SELECT x, tstate(avg(y)) FROM stream GROUP BY x;

    -- combiner
    SELECT x, tcombine(avg, tstate) FROM worker GROUP BY x;

The reason this came up during work on #1068 is that the outer `array_agg` query on the worker needs access to the inner transition state, which currently isn't possible because the tstate columns aren't generated until much later, so things would be messy without this refactor.

**Comment by usmanm** [comment]
To clarify a bit, I think this would be:

```
-- worker
SELECT x, tstate(avg(y)) AS avg FROM stream GROUP BY x;

-- combiner
SELECT x, tcombine(avg) AS avg FROM (worker UNION matrel) GROUP BY x;

-- view 
SELECT x, transout(avg) AS avg FROM matrel;
```

**Comment by derekjn** [comment]
The `avg` in `tcombine` is actually the function name to combine with, as the combiner query doesn't know anything about what function was used to generate that transition state on the worker (at least it shouldn't). In practice it would be an `oid`: `tcombine(5020, worker_tstate)`.

**Comment by derekjn** [comment]
Also, user combines would essentially be rewritten to: `finalize(tcombine(<oid>, tstate))`.

**Comment by usmanm** [comment]
I thought `tcombine` is just a placeholder for the combine function for that specific aggregate? I'm not sure why the combiner query shouldn't know what function to use. Both the worker and combiner query know which aggregate is being used. It's just a matter of picking up the right function from the `pipeline_combine` catalog then. Worker picks the tstate *accum* function, and combiner picks the tstate *combine* function.

Yes, good call, user combines will have to be re-written as that. Same comment about passing in OID of the combine func here though.

**Comment by derekjn** [comment]
That will also work. I guess my thinking is to move away as much as possible from special cases for continuous query plans. e.g. so something like this would technically work:

    SELECT tcombine(<avg oid>, tstate) FROM (SELECT '{1, 2}' AS tstate)

It's probably best to just stick with picking the transition, combine, and finalize functions for simplicity's sake though. I'll do that.

**Comment by derekjn** [comment]
Merged into the #1032 branch.
